### ISSUE TITLE
A11y: switch requires aria-checked on document

### ISSUE THREAD

**Issue by abiyasa** [eBay/skin#832]
I got a11y errors when I tested the Skin doc/demo page for DS6 using [Accessibility Insight](https://accessibilityinsights.io/) plugin from Microsoft.

![a11y-insight-switch](https://user-images.githubusercontent.com/1072047/64949147-d9026a00-d878-11e9-837e-f12b7911a4fb.png)

It complains that `aria-checked` is missing. I checked again [wai-aria-1.1 about switch role](https://www.w3.org/TR/wai-aria-1.1/#switch), it's true that we need to have `aria-checked` attribute if we set the role to switch.

Here's details from accessibility insights report:

```
Title: WCAG 4.1.1,WCAG 4.1.2: Required ARIA attributes must be provided (#field-unstacked-3)
Tags: Accessibility, WCAG 4.1.1, WCAG 4.1.2, aria-required-attr

Issue: Required ARIA attributes must be provided (aria-required-attr - https://dequeuniversity.com/rules/axe/3.2/aria-required-attr?application=msftAI)

Target application: eBay Skin - for DS6 - https://ebay.github.io/skin/ds6/index.html#switch

Element path: #field-unstacked-3

Snippet: <input aria-label="Switch Demo" class="switch__control" role="switch" type="checkbox" id="field-unstacked-3">

How to fix: 
Fix any of the following:
  Required ARIA attribute not present: aria-checked

Environment: Chrome version 77.0.3865.75

====

This accessibility issue was found using Accessibility Insights for Web 2.9.2 (axe-core 3.2.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.
```

**Comment by ianmcburnie** [comment]
The actual user impact may be zero here. If you compare checkbox with switch in Safari+VO, you will notice that checkbox announces as "Checked/unchecked", switch announces as "On/off". The role of switch is piggy-backing off of the checked/unchecked state of the input.

The benefit of the switch the way it is currently is coded, is that no JavaScript is required (to manage aria-checked state). However, I've actually been thinking about rewriting switch so that it is a  pure ARIA/JavaScript widget, and more in line with ARIA 1.1 (our switch was coded a long time ago). Using checkbox under the hood is a neat trick to keep it pure CSS, but a switch doesn't submit form data (it updates instantly) so the input element is redundant really. Now that ARIA support is in place, it can, and maybe should, be a div.

I will probably leave things as they are for now but will consider making the change in next major version.

**Comment by abiyasa** [comment]
Thanks! I wanted to suggest removing `role="switch"` so switch is basically a skinned checkbox, but you're right, it's much better & makes more sense for the VO to announce switch as an on/off rather than checked/unchecked.

I agree with you to remove the input element at some point since switch updates instantly while checkbox submits form data. We might lose the progressive enhancement but for the instant update functionality, we need JS anyway.

**Comment by ianmcburnie** [comment]
I'm a big fan of Accessibility Insights, so after more thought, I'm going to go ahead now and add the tiny bit of javascript to keep it happy. We can do the same on the CoreUI side. Thanks for raising this @abiyasa

**Comment by abiyasa** [comment]
No problem! If you want, I can also help you update the CoreUI switch component.

I actually found this issue when we were porting CoreUI Switch Marko component to React. We use Storybook with [a11y add-on](https://www.npmjs.com/package/@storybook/addon-a11y), that's when I found this error message. I thought it was our fault while porting the component so I checked the Skin doc and ran Accessibility Insights on it.

**Comment by ianmcburnie** [comment]
Yes, feel free to send a PR there. Thanks!
